src/Tutor.Core/DomainModel/AssessmentEvents/SubmissionService.cs

```diff
@@ -29,11 +29,7 @@ public Result<Evaluation> EvaluateAndSaveSubmission(Submission submission)
 
             var result = knowledgeComponentMastery.SubmitAEAnswer(submission);            
 
-            if (result.IsSuccess)
-            {
-                _kcRepository.UpdateKCMastery(knowledgeComponentMastery);
-                _assessmentEventRepository.SaveSubmission(submission);
-            }
+            _kcRepository.UpdateKCMastery(knowledgeComponentMastery);
```

Comment:
I take it we removed the IF since UpdatingKCMastery won't do anything if the result is unsuccessful?

Reply:
The if was there because the Submission shouldn't be added if evaluation fails. I moved adding the Submission to the When method, so KcMastery now handles if it's getting added and where. And calling the UpdateKCMastery is fine either way, since it always keeps itself in a valid state when its methods are called.